### PR TITLE
ENH: tools for updating IOCs, more docs, and current configuration

### PR DESCRIPTION
**Pull request by klauer**
Background
------------
I'd like to get closer to finishing https://github.com/pcdshub/Bug-Reports-and-Requests/issues/20

XCS IOC is currently deployed thanks to @ZryletTC. XPP one is not yet updated, and I don't think the current generated config includes the beryllium lens stack PVs of https://github.com/pcdshub/Bug-Reports-and-Requests/issues/48.

Adds
------
* Tools for updating IOCs more easily
* More documentation in the form of README.md
* Current generated configuration from the dev environment

How to update
----------

Overview:

```bash
$ cd config
$ make clean all iocs
$ cd iocs/xcs/
$ epics-release ...
```

Questions
-----------
* As the only current user, @ZryletTC, does the XCS config look OK?
* Can the `epics-release` portion be made any easier with an auto version-bumping tool, or should this be left manual? Any other suggestions on the overall process are welcome of course, too.

### CODE REVIEW

**Comment by ZLLentz** [comment]
RE: the second point in your questions, doing an `epics-release` automatically here would feel a bit weird because the ioc directory we'd be releasing is shared between all pv notepad pvs for the particular hutch... a separate `make epics-release` command may be useful though? I haven't thought about this for long enough to consider all the avenues.

The manual release is a one-liner so the point might be moot.

File: config/Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-beamlines            := cxi xcs xpp mec mfx hxd icl kfe lfe mec mfx pbt tmo
+beamlines            := cxi xcs xpp mec mfx hxd icl kfe lfe tmo
```
Comment (ZLLentz) [inline config/Makefile]: it might be time to add rix

File: config/iocs/Makefile [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+all:
+	@echo "Checking out and updating IOCs..."
+	@for fn in ../notepad/*.cfg; do \
+		hutch=$$(echo $$fn | sed s/.*ioc-// | sed s/-pcdsdevices.cfg// ); \
+		echo "Hutch: $$hutch"; \
+		ioc_path=$$EPICS_REPO/epics/trunk/ioc/$$hutch/pvNotepad/current; \
+		if (svn ls $$ioc_path > /dev/null 2>&1); then \
+			echo "* Found IOC in SVN"; \
+			svn checkout $$ioc_path $$hutch || svn resolve --accept theirs-full $$hutch; \
```
Comment (ZLLentz) [inline config/iocs/Makefile:11]: This is a godsend for people like me who can't remember more than one VCS at a time

**Review by ZLLentz — APPROVED** [review]
I think this looks solid

File: config/Makefile (hunk 1 above)
Comment (klauer) [reply config/Makefile]: Oops, good call - resolved in da897ce

File: config/iocs/Makefile (hunk 2 above)
Comment (klauer) [reply config/iocs/Makefile:11]: I'm also guilty of this - working back from "_Git for SVN users cheatsheet_" is my new go-to...

**Comment by ZryletTC** [comment]
Wow that's a lot of files, thanks for getting this going! I'll look through them more tomorrow, but some preliminary comments for now:
- The XCS file mostly looks good but I believe I had also added a couple lines manually for devices that are set up in the beamline files. For XCS it looks like that is two PVs for the SND system and for XPP, there are some extra lxt PVs.
- As @ZLLentz was saying, automatic IOC update might be a little weird and I think I'd rather go the manual route for deployment/release. One reason is for any extra PVs above, but primarily because the SVN structure is a little weird and different than the list of beamlines that happi has (i.e. Each of these configs live down in separate subdirectories and there's no hxd subdirectory as far as I'm aware). It might be worth reorganizing a bit :/
- As for the auto-version-bumping, that would be a nice feature but I think it should live in the epics-release script itself (at the cost of feeding the beast of a script it is already)
- Can the non-pretty-printed JSONs be pretty printed, pretty please?

Edit: Maybe nvm on those top two points, it looks like you've done a lot of working through that already. I'll have to read through it more and get back to you.

**Comment by klauer** [comment]
> RE: Version-bumping

OK, I think my lack of IOC deployment knowledge got in the way here. It seems that doing it manually is the right way.

> Can the non-pretty-printed JSONs be pretty printed, pretty please?

Good call! Done.

Let me know when you get a chance on the first two points, @ZryletTC. No real rush.

**Comment by klauer** [comment]
@ZryletTC I forgot about this PR, but apparently I had asked you about something in the lengthy text above. Should we move forward here?

**Review by ZryletTC — APPROVED** [review]
Hey sorry, I kinda forgot about this as well. Yes, let's move forward with this, the code is all good 👍
My only request is please don't deploy to XCS or XPP without me.

**Pull request merged**